src/main/java/org/mockito/internal/configuration/plugins/PluginLoader.java

```diff
@@ -51,20 +59,22 @@ public PluginLoader(PluginSwitch pluginSwitch) {
         try {
             resources = loader.getResources("mockito-extensions/" + service.getName());
         } catch (IOException e) {
-            throw new MockitoException("Failed to load " + service, e);
+            // use RuntimeException instead MockitoException, because MockitoException depends on plugins and can
+            // break stack track
+            throw new RuntimeException("Failed to load " + service, e);
         }
 
         try {
             String foundPluginClass = new PluginFinder(pluginSwitch).findPluginClass(Iterables.toIterable(resources));
             if (foundPluginClass != null) {
-                Class<?> pluginClass = loader.loadClass(foundPluginClass);
+                Class<?> pluginClass = Class.forName(foundPluginClass);
```

Comment:
What do you think of trying first `loader.loadClass(found)` then if it doesn't work try the `Class.forName(found)` ?
I'm wary of the plain `Class.forName` as it will always choose the mockito current classloader.